libs/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts

```diff
@@ -68,15 +66,20 @@ export class DataLayerColorService {
   /** Finds a matching dataset with similar label and returns its color index in the palette */
   private getMatchingDatasetColorIndex(layer: DataLayer, dataset: Dataset) {
     for (let other of layer.datasets) {
-      if (isMatchingDataset(dataset, other)) {
+      if (dataset.chartsOnFhir?.group && dataset.chartsOnFhir.group === other.chartsOnFhir?.group) {
         const color = this.getColor(other);
-        if (color) {
-          for (let palette of [this.palette, this.lightPalette]) {
-            const index = palette.indexOf(color);
-            if (index >= 0) {
-              return index;
-            }
-          }
+        return this.datasetColorIndex(color!);
+      }
+    }
+    return null;
+  }
+
+  private datasetColorIndex(color: string) {
+    if (color) {
+      for (let palette of [this.palette, this.lightPalette]) {
+        const index = palette.indexOf(color);
+        if (index >= 0) {
+          return index;
         }
       }
     }
```

Comment:
You made a subtle change to the behavior of this function when you factored out `datasetColorIndex()`.  `getMatchingDatasetColorIndex()` now returns undefined if the first dataset in the group does not have a color defined (or if it's not in either of the palettes). Previously, it would move on to check the next dataset in the group and would only return null after checking all of them.